src/py_zfs_pool.c

```diff
@@ -28,378 +24,252 @@ int py_zfs_pool_init(PyObject *type, PyObject *args, PyObject *kwds) {
+	if (self == NULL)
+		Py_RETURN_NONE;
+	return Py_NewRef(self->name);
 }
 
 static
-PyObject *py_zfs_pool_sync(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_create(PyObject *self, PyObject *args) {
+PyObject *py_zfs_pool_asdict(PyObject *self, PyObject *args) {
 	Py_RETURN_NONE;
 }
 
 static
-PyObject *py_zfs_pool_attach_vdevs(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+PyObject *py_zfs_pool_root_dataset(PyObject *self, PyObject *args) {
+	PyObject *out = NULL;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	PyObject *open_rsrc;
+	PyObject *fargs = NULL;
+	PyObject *fkwargs = NULL;
 
-static
-PyObject *py_zfs_pool_sync_vdev_by_guid(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	fargs = PyTuple_New(0);
+	if (fargs == NULL)
+		return (NULL);
 
-static
-PyObject *py_zfs_pool_delete(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	fkwargs = Py_BuildValue("{s:O}", "name", p->name);
+	if (fkwargs == NULL) {
+		Py_DECREF(fargs);
+		return (NULL);
+	}
 
-static
-PyObject *py_zfs_pool_start_scrub(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
+	open_rsrc = PyObject_GetAttrString((PyObject*)p->pylibzfsp,
+					   "open_resource");
+	out = PyObject_Call(open_rsrc, fargs, fkwargs);
 
-static
-PyObject *py_zfs_pool_stop_scrub(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
+	Py_DECREF(open_rsrc);
+	Py_DECREF(fargs);
+	Py_DECREF(fkwargs);
+	return (out);
 }
 
 static
 PyObject *py_zfs_pool_clear(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_upgrade(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_ddt_prefetch(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_ddt_prune(PyObject *self, PyObject *args) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_root_dataset(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_root_vdev(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_data_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_log_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_cache_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_spare_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_special_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_dedup_vdevs(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
-
-static
-PyObject *py_zfs_pool_get_groups(py_zfs_prop_t *self, void *extra) {
-	Py_RETURN_NONE;
-}
+	int ret = 0;
+	nvlist_t *policy = NULL;
+	py_zfs_pool_t *p = (py_zfs_pool_t *)self;
+	py_zfs_error_t err;
+
+	Py_BEGIN_ALLOW_THREADS
+	policy = fnvlist_alloc();
```

Comment:
If we don't need to check for NULL here, needs comment for future devs why :)

Reply:
Added a comment:
```
	/* Force nvlist wrapper wraps the nvlist_* functions with assertions
	 * that assume the operation is successful. We can avoid checking for
	 * NULL here since as assert would be hit from libspl in case the
	 * operation failed.
	 */
```

Reply:
Looks great